ludwig/models/llm.py

```diff
@@ -46,12 +46,16 @@ def __init__(
 
         logger.info("Loading large language model...")
         self.model = AutoModelForCausalLM.from_pretrained(self.config_obj.model_name)
+        ####SXK should we make this configurable ######
```

Comment:
This is actually a requirement when using the Ray backend. In theory one could directly load it onto GPUs, but the issue here is that Ray isn't aware of how many GPUs are available to it until later, so we load it onto CPU first and then move it onto GPUs in a later function (to_device). 

In the future, we plan to support meta tensor based initialization since this will not actually load the weights onto the CPU, but just the shapes. Then, if we're doing model parallel training, it'll move the tensors onto the right GPU/CPU devices when it needs to, along wit the optimizer and activations.